x-pack/legacy/plugins/snapshot_restore/public/app/components/snapshot_delete_provider.tsx

```diff
@@ -0,0 +1,194 @@
+              {
+                defaultMessage: 'Deleted {count} snapshots',
+                values: { count: itemsDeleted.length },
+              }
+            )
+          : i18n.translate('xpack.snapshotRestore.deleteSnapshot.successSingleNotificationTitle', {
+              defaultMessage: "Deleted snapshot '{name}'",
+              values: { name: itemsDeleted[0].snapshot },
+            });
+        toastNotifications.addSuccess(successMessage);
+        if (onSuccessCallback.current) {
+          onSuccessCallback.current([...itemsDeleted]);
+        }
+      }
+
+      // Surface error notifications
+      // `error` is generic server error
+      // `data.errors` are specific errors with removing particular snapshot(s)
+      if (error || (errors && errors.length)) {
+        const hasMultipleErrors =
+          (errors && errors.length > 1) || (error && snapshotsToDelete.length > 1);
+        const errorMessage = hasMultipleErrors
+          ? i18n.translate('xpack.snapshotRestore.deleteSnapshot.errorMultipleNotificationTitle', {
+              defaultMessage: 'Error deleting {count} snapshots',
+              values: {
+                count: (errors && errors.length) || snapshotsToDelete.length,
+              },
+            })
+          : i18n.translate('xpack.snapshotRestore.deleteSnapshot.errorSingleNotificationTitle', {
+              defaultMessage: "Error deleting snapshot '{name}'",
+              values: { name: (errors && errors[0].id.snapshot) || snapshotsToDelete[0].snapshot },
+            });
+        toastNotifications.addDanger(errorMessage);
+      }
+    });
+  };
+
+  const renderModal = () => {
+    if (!isModalOpen) {
+      return null;
+    }
+
+    const isSingle = snapshotIds.length === 1;
+
+    return (
+      <EuiOverlayMask>
+        <EuiConfirmModal
+          title={
+            isSingle ? (
+              <FormattedMessage
+                id="xpack.snapshotRestore.deleteSnapshot.confirmModal.deleteSingleTitle"
+                defaultMessage="Delete snapshot '{name}'?"
+                values={{ name: snapshotIds[0].snapshot }}
+              />
+            ) : (
+              <FormattedMessage
+                id="xpack.snapshotRestore.deleteSnapshot.confirmModal.deleteMultipleTitle"
+                defaultMessage="Delete {count} snapshots?"
+                values={{ count: snapshotIds.length }}
+              />
+            )
+          }
+          onCancel={closeModal}
+          onConfirm={deleteSnapshot}
+          cancelButtonText={
+            <FormattedMessage
+              id="xpack.snapshotRestore.deleteSnapshot.confirmModal.cancelButtonLabel"
+              defaultMessage="Cancel"
+            />
+          }
+          confirmButtonText={
+            isSingle ? (
+              <FormattedMessage
+                id="xpack.snapshotRestore.deleteSnapshot.confirmModal.confirmSingleButtonLabel"
+                defaultMessage="Delete snapshot"
```

Comment:
The plural syntax here works much better (also applied for other labels using `isSingle`)

```
	defaultMessage="Delete {snapshotCount, plural, one {snapshot} other {snapshots}}"
	values={{snapshotCount: snapshotIds.length}}
```